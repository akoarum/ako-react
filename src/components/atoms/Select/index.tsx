import styled from 'styled-components'
import * as React from 'react'
import * as style from '../../../utils/style'
import { Choice } from '../../../models/types'

type Props = {
  className?: string
  name: string
  options: Choice[]
  value: string | number
  id?: string
  required?: boolean
  onChange?: (e: React.FormEvent<HTMLSelectElement>) => void
  onFocus?: (e: React.FormEvent<HTMLSelectElement>) => void
  onBlur?: (e: React.FormEvent<HTMLSelectElement>) => void
}

export const StyledRoot = styled.div`
  position: relative;
  display: inline-block;

  &::after {
    position: absolute;
    top: 50%;
    right: 12px;
    content: '';
    margin-top: -4px;
    border-style: solid;
    border-color: ${style.color.theme} transparent transparent;
    border-width: 8px 5px;
    pointer-events: none;
  }

  ${style.mediaQuery.pc} {
    min-width: 240px;
  }

  ${style.mediaQuery.sp} {
    width: 100%;
  }

  .select__content {
    display: inline-block;
    box-sizing: border-box;
    border-radius: 0;
    box-sizing: border-box;
    border: 1px solid ${style.color.border};
    padding: 8px 32px 8px 12px;
    width: 100%;
    font-size: ${style.font.sizePc};
    line-height: 1.3;
  }
`

export const Select: React.FC<Props> = props => {
  const Options = props.options.map(option => {
    const value = option.value ? option.value : option.id
    return (
      <option key={option.id} value={value}>
        {option.label}
      </option>
    )
  })
  return (
    <StyledRoot className={props.className}>
      <select
        name={props.name}
        value={props.value}
        id={props.id}
        required={props.required}
        className="select__content"
        onChange={props.onChange}
        onFocus={props.onFocus}
        onBlur={props.onBlur}
      >
        {Options}
      </select>
    </StyledRoot>
  )
}

export default Select
