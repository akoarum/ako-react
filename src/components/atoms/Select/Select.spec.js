import React from 'react'
import renderer from 'react-test-renderer'
import Select from './index'

describe('Select', () => {
  let wrapper
  const options = [
    { id: 1, label: '項目1' },
    { id: 2, label: '項目2' },
    { id: 3, label: '項目3' }
  ]

  beforeEach(() => {
    wrapper = renderer.create(
      <Select name="name" options={options} value="" />
    )
  })

  describe('initialize', () => {
    it('snapshot', () => {
      const tree = wrapper.toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
