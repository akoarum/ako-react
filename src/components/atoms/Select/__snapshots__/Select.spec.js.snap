// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select initialize snapshot 1`] = `
<div
  className="sc-bdVaJa gtKVFJ"
>
  <select
    className="select__content"
    name="name"
    value=""
  >
    <option
      value={1}
    >
      項目1
    </option>
    <option
      value={2}
    >
      項目2
    </option>
    <option
      value={3}
    >
      項目3
    </option>
  </select>
</div>
`;
